### PR TITLE
New branch for Netcat improvements and formatted

### PR DESCRIPTION
**Pull request by basoreeper3007**
### Bugs/Issues:
-Port Range Scan not working due to insufficient input handling.
-Send File not working (incorrect syntax and implementation).
-Receive File not working (incorrect syntax and implementation).

### Fixes:
-Port Range Scan: Added logic that checks if a "-" is present in the port input. If a "-" is present it splits the string into a start and end port range. The port scan command is then run iteratively from the start port to the end port.
-Send File: Made the code more simple by reducing LOC. Changed the syntax of the nc command. Altered the way the command was executed through command helper by running the nc command through a bash shell.
-Receive File: Fixes for the receive file feature are similar to send file, only difference is the syntax of the nc command.
-Comments: I wrote sufficient comments to explain the code for each feature, for more information about the features see the code comments.

### CODE REVIEW

**Comment by zkkok** [comment]
Hi Bas,
Thanks for putting in such a great effort. The coding side looks good with no apparent logical problems. Waiting for @gangamarx to perform the functionality test and your code is good to be merged.

Just a suggestion on future issues that you will be working on, instead of writing one commit for the whole block of code, it is better to break it down into small parts each with their own commits, this makes it easier for reviewers to review your code and for you to understand when you come back to the code in the future. Here is a guide on how to write commit messages shared by @bradleybogg [https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/](url)

**Comment by bradleybogg** [comment]
@gangamarx to conduct application testing. Upon review I will review code.

File: src/components/Netcat/Netcat.tsx
Comment (bradleybogg) [inline src/components/Netcat/Netcat.tsx:1]: Requires Application testing.

@gangamarx  please test this component.

Thank you!

**Review by bradleybogg — CHANGES_REQUESTED** [review]

**Comment by zkkok** [comment]
Hi @basoreeper3007,
Functionality testing has been conducted. Here are the details:
[Functionality testing(Netcat).pdf](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/files/15158826/Functionality.testing.Netcat.pdf)

All intended functionalities work well. However here are some suggestion if you are looking to improve the code before merging:

1. Rework the scan range of ports function to show summary of all ports (Test#1)
2. Files outside of ddt can be transferred also, just needs specifying file path (Update comment for 
send file operation)

4. Add LoadingOverlayAndCancel Component to properly terminate processes and show a more 
indicative sign that the program is running
5. Adding the option to select verbose mode (-v) and timeout (-w) rather than hardcoding it 
6. Combine website port scan and port scan since they do similar things and run on similar
commands
7. Remove/Fix “Interactive terminal” function
8. Hide irrelevant fields to a specific function. (Ex. When selecting “Receive files” only port number 
and File Path is required, hide other irrelevant fields

**Comment by basoreeper3007** [comment]
Hey @zkkok, 

I plan to move to a new tool as I've spent a good amount of time on Netcat and want my technical contributions to span over another tool as well. I will give it a go doing some of the suggestions you provided, however I likely won't do them all. 

I will likely do suggestions:

2. Files outside of ddt can be transferred also, just needs specifying file path (Update comment for
send file operation)

4. Adding the option to select verbose mode (-v) and timeout (-w) rather than hardcoding it

7. Hide irrelevant fields to a specific function. (Ex. when selecting “Receive files” only port number
and File Path is required, hide other irrelevant fields

I would also like to note that "6. Remove/Fix “Interactive terminal” function" has been done in another netcat branch im working on where im implementing a Listen feature for Netcat, so this change will be apparent in my next pull request.

**Comment by basoreeper3007** [comment]
@zkkok and @bradleybogg 

I've done suggestions 2, 6 ,7 and 4 (with 4 i didn't implement an option for timeout as its only used for the send feature and isn't really important for the user to be able to alter the timeout time)

I'm finished with this branch. Once this is merged into main i will merge main with my other netcat branch (502-feature-request-netcat-listen-and-connect), and pull request those changes for my Listen feature.

**Review by bradleybogg — APPROVED** [review]

**Comment by bradleybogg** [comment]
@basoreeper3007 

Good work. There is some work here that needs to be completed. I have added these to https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/issues/598 baseline issue report.

**Pull request merged**